### PR TITLE
SDKS-3642 Enable Dynamic Environment Switching in Test App

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[SDKS-3642](https://pingidentity.atlassian.net/browse/SDKS-3642)

# Description

SDKS-3642 Enable Dynamic Environment Switching in Test App

### CODE REVIEW

**Review by spetrov — APPROVED** [review]
LGTM! 👍🏻

File: samples/app/src/main/java/com/pingidentity/samples/app/Setting.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2024 PingIdentity. All rights reserved.
```
Comment (spetrov) [inline samples/app/src/main/java/com/pingidentity/samples/app/Setting.kt:2]: Typo in PingIdentity (missing space...)

File: samples/app/src/main/java/com/pingidentity/samples/app/davinci/DaVinciViewModel.kt [hunk 2]
```diff
@@ -9,13 +9,10 @@ package com.pingidentity.samples.app.davinci
```
Comment (spetrov) [inline samples/app/src/main/java/com/pingidentity/samples/app/davinci/DaVinciViewModel.kt:9]: Typo in the copyright header - missing space in `PingIdentity`...

File: samples/app/src/main/java/com/pingidentity/samples/app/env/Env.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 PingIdentity. All rights reserved.
```
Comment (spetrov) [inline samples/app/src/main/java/com/pingidentity/samples/app/env/Env.kt:2]: Typo in the copyright header - missing space in `PingIdentity`...

File: samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 PingIdentity. All rights reserved.
```
Comment (spetrov) [inline samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt:2]: Typo in the copyright header - missing space in `PingIdentity`...

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍🏻 
The copyright header in couple of files need to be updated though...

File: samples/app/src/main/java/com/pingidentity/samples/app/davinci/DaVinciViewModel.kt (hunk 2 above)
Comment (witrisna) [reply samples/app/src/main/java/com/pingidentity/samples/app/davinci/DaVinciViewModel.kt:9]: Updated

File: samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import com.pingidentity.samples.app.User
+import com.pingidentity.samples.app.settingDataStore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+
+val test = DaVinci {
+
+    logger = Logger.STANDARD
```
Comment (rodrigoareis) [inline samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt:33]: Can we remove those configs and let it to just rely on whatever the developer has on their local? We don't want external developers having details of our environments

**Review by rodrigoareis — COMMENTED** [review]
Changes looks good to me, minor comment to address

**Review by jeyanthanperiyasamy — APPROVED** [review]

File: samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt (hunk 5 above)
Comment (witrisna) [reply samples/app/src/main/java/com/pingidentity/samples/app/env/EnvViewModel.kt]: Will keep this in mind, and will update it in another story.

**Pull request merged**